### PR TITLE
Fix layout context prop resolving for boolean types (#567)

### LINKED ISSUE #567: Layout context not working for bool props
**Usecase:**

Im trying to get an `InputGroup` with one child element disabled and one not:

```jsx
<InputGroup>
  <TextField
    label="First name"
    disabled
  />
  <TextField
    label="Last name"
  />
</InputGroup>
```

**Problem:**

In the above example:

Current behaviour:  none of the `TextField`s is disabled
Expected behaviour: The `First name`  `TextField` is disabled

Likely cause is https://github.com/react-ui-org/react-ui/blob/d532592001380c4c88a0f92d637e1b6bfbead5ce/src/components/_helpers/resolveContextOrProp.js#L2 where the `contextProp` is checked for being `null` / `undefined` but not against being `false`.

**Solution:**

We need to figure out how to take into account the default prop value in https://github.com/react-ui-org/react-ui/blob/d532592001380c4c88a0f92d637e1b6bfbead5ce/src/components/_helpers/resolveContextOrProp.js#L2. Assuming the default is `null` / `undefined` does not work for bool typed props.

### PR DESCRIPTION
**Pull request by atmelmicro**
Allows boolean props to propagate when layout context is false

Closes #567

### CODE REVIEW

File: src/components/_helpers/resolveContextOrProp.js [hunk 1]
```diff
@@ -1,4 +1,8 @@
 export const resolveContextOrProp = (contextValue, propValue) => {
+  if (contextValue === false) {
```
Comment (mbohal) [inline src/components/_helpers/resolveContextOrProp.js:2]: Could you please rewrite it so there are only two returns?

```
if (…) {
  return a;
}

return b;
```

Having three returns makes it hard to follow the flow of the code. I know this function is short, but still…

**Review by mbohal — CHANGES_REQUESTED** [review]

File: src/components/_helpers/resolveContextOrProp.js
Comment (adamkudrna) [inline src/components/_helpers/resolveContextOrProp.js:1]: Does it make any sense to have a test for it? Possibly with an example similar to that in the original issue #567.

Comment (mbohal) [reply src/components/_helpers/resolveContextOrProp.js:1]: I think Ill leave it up to @atmelmicro to decide.

It is one simple condition, so I think that maybe just a comment would work:

```js
// We need to test:
//  * `false` - for when the `contextValue` is boolean
//  * `null` - for when the `contextValue` is non-boolean
if (contextValue === false || contextValue === null) {
```

However test would communicate it just as good and it would test it on top.

I have no preference, really, but I agree with @adamkudrna that one or the other should be added.

Comment (atmelmicro) [reply src/components/_helpers/resolveContextOrProp.js:1]: In my opinion the comment is more descriptive, it describes the why, not just how it should work.

**Review by mbohal — APPROVED** [review]

**Review by bedrich-schindler — APPROVED** [review]
